internal/pspdriver/pspdriver.go

```diff
@@ -0,0 +1,318 @@
+	GuestSVN         uint32
+	Policy           uint64
+	FamilyID         [16]byte
+	ImageID          [16]byte
+	VMPL             uint32
+	SignatureAlgo    uint32
+	PlatformVersion  uint64
+	PlatformInfo     uint64
+	AuthorKeyEn      uint32
+	Reserved1        uint32
+	ReportData       [SnpPspReportDataSize]byte
+	Measurement      [48]byte
+	HostData         [SnpPspReportHostDataSize]byte
+	IDKeyDigest      [48]byte
+	AuthorKeyDigest  [48]byte
+	ReportID         [32]byte
+	ReportIDMA       [32]byte
+	ReportTCB        uint64
+	Reserved2        [24]byte
+	ChipID           [64]byte
+	CommittedSVN     [8]byte
+	CommittedVersion [8]byte
+	LaunchSVN        [8]byte
+	Reserved3        [168]byte
+	Signature        [512]byte
+}
+
+// Report represents parsed attestation report.
+type Report struct {
+	Version          uint32
+	GuestSVN         uint32
+	Policy           uint64
+	FamilyID         string
+	ImageID          string
+	VMPL             uint32
+	SignatureAlgo    uint32
+	PlatformVersion  uint64
+	PlatformInfo     uint64
+	AuthorKeyEn      uint32
+	ReportData       string
+	Measurement      string
+	HostData         []byte
+	IDKeyDigest      string
+	AuthorKeyDigest  string
+	ReportID         string
+	ReportIDMA       string
+	ReportTCB        uint64
+	ChipID           string
+	CommittedSVN     string
+	CommittedVersion string
+	LaunchSVN        string
+	Signature        string
+}
+
+func (sr *report) report() Report {
+	return Report{
+		Version:          sr.Version,
+		GuestSVN:         sr.GuestSVN,
+		Policy:           sr.Policy,
+		FamilyID:         hex.EncodeToString(mirrorBytes(sr.FamilyID[:])[:]),
+		ImageID:          hex.EncodeToString(mirrorBytes(sr.ImageID[:])[:]),
+		VMPL:             sr.VMPL,
+		SignatureAlgo:    sr.SignatureAlgo,
+		PlatformVersion:  sr.PlatformVersion,
+		PlatformInfo:     sr.PlatformInfo,
+		AuthorKeyEn:      sr.AuthorKeyEn,
+		ReportData:       hex.EncodeToString(sr.ReportData[:]),
+		Measurement:      hex.EncodeToString(sr.Measurement[:]),
+		HostData:         sr.HostData[:],
+		IDKeyDigest:      hex.EncodeToString(sr.IDKeyDigest[:]),
+		AuthorKeyDigest:  hex.EncodeToString(sr.AuthorKeyDigest[:]),
+		ReportID:         hex.EncodeToString(sr.ReportID[:]),
+		ReportIDMA:       hex.EncodeToString(sr.ReportIDMA[:]),
+		ReportTCB:        sr.ReportTCB,
+		ChipID:           hex.EncodeToString(sr.ChipID[:]),
+		CommittedSVN:     hex.EncodeToString(sr.CommittedSVN[:]),
+		CommittedVersion: hex.EncodeToString(sr.CommittedVersion[:]),
+		LaunchSVN:        hex.EncodeToString(sr.LaunchSVN[:]),
+		Signature:        hex.EncodeToString(sr.Signature[:]),
+	}
+}
+
+// mirrorBytes mirrors the byte ordering so that hex-encoding little endian
+// ordered bytes come out in the readable order.
+func mirrorBytes(b []byte) []byte {
+	for i := 0; i < len(b)/2; i++ {
+		mirrorIndex := len(b) - i - 1
+		b[i], b[mirrorIndex] = b[mirrorIndex], b[i]
+	}
+	return b
+}
+
+var (
+	amdsnppspapi = windows.NewLazySystemDLL("amdsnppspapi.dll")
```

Comment:
If you want to call a win32 API, please add it under `internal/winapi` (in a separate file if necessary) and then run `go generate` on that package to automatically generate the go code for calling that API. Look at other files under `internal/winapi` to see the format in which you would need to add your methods.

Reply:
Addressed it in https://github.com/microsoft/hcsshim/pull/2518/commits/2f21484a4dd2ff4fe500c6eb9d169a8efc854bb4